### ISSUE TITLE
[Usage] loss is 0.0 and tokenization mismatch: 102 vs. 104. (ignored)  on liuhaotian/llava-v1.6-34b 

### ISSUE THREAD

**Issue by awzhgw** [haotian-liu/LLaVA#1166]
### Describe the issue

When I use the liuhaotian/llava-v1.6-34b model to fine-tune my own data, I encounter an alarm: tokenization mismatch: 102 vs. 104.
how to resolve it ?

my start shell is :

```shell
nohup deepspeed llava/train/train_mem.py \
    --lora_enable True --lora_r 128 --lora_alpha 256 --mm_projector_lr 2e-5 \
    --deepspeed ./scripts/zero3.json \
    --model_name_or_path ${CheckPoint_ROOT}/${ModelName} \
    --version v1 \
    --data_path ${JSON_PATH} \
    --image_folder ${IMAGE_FOLDER} \
    --vision_tower ${VISION_TOWER} \
    --mm_projector_type mlp2x_gelu \
    --mm_vision_select_layer -2 \
    --mm_use_im_start_end False \
    --mm_use_im_patch_token False \
    --image_aspect_ratio pad \
    --group_by_modality_length True \
    --bf16 True \
    --output_dir ${CheckPoint_ROOT}/${ModelName}-lora \
    --num_train_epochs 1 \
    --per_device_train_batch_size 16 \
    --per_device_eval_batch_size 4 \
    --gradient_accumulation_steps 1 \
    --evaluation_strategy "no" \
    --save_strategy "steps" \
    --save_steps 50000 \
    --save_total_limit 1 \
    --learning_rate 2e-4 \
    --weight_decay 0. \
    --warmup_ratio 0.03 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --tf32 True \
    --model_max_length 2048 \
    --gradient_checkpointing True \
    --dataloader_num_workers 4 \
    --lazy_preprocess True \
    --report_to tensorboard  > ${Local_DATA_ROOT}/llava_tune.log 2>&1 &
```

my data is like
```
[
  {
    "id": "100069386727",
    "image": "self-operated-sku/11/d13f9a7286fe4d0be216be27434a9a70.jpg",
    "conversations": [
      {
        "from": "human",
        "value": "<image>\n这张图片里的商品信息都有哪些？"
      },
      {
        "from": "gpt",
        "value": "图片中的商品名称为 公牛（BULL）新国标插座/插线板/插排/排插/接线板/拖线板GN-6074位总控全长4米。品牌名称是 公牛。分类信誉属于 电脑、办公,外设产品,插座。"
      }
    ]
  },
  {
    "id": "100047854046",
    "image": "self-operated-sku/61/077c741f29180fd58bbc76fd035b8f68.jpg",
    "conversations": [
      {
        "from": "human",
        "value": "<image>\n这张图片里都描述了什么信息？"
      },
      {
        "from": "gpt",
        "value": "图片中的商品名称为 得力(deli)笔记本子皮面本活页记事本可换芯带插袋【皮面套装】(黑色)(盒)22296。品牌名称是 得力（deli）。分类信誉属于 电脑、办公,文具,本册/便签。"
      }
    ]
  },
  {
    "id": "100055983055",
    "image": "self-operated-sku/19/373ddb22228b05bb0edfddfe44609df5.jpg",
    "conversations": [
      {
        "from": "human",
        "value": "<image>\n请列举出这张图片中商品的相关信息？"
      },
      {
        "from": "gpt",
        "value": "图片中的商品名称为 公牛（BULL）新国标插座/插线板/插排/排插/接线板/拖线板GN-2114位总控全长1.8米NG111。品牌名称是 公牛。分类信誉属于 工业品,中低压配电,插座。"
      }
    ]
  },
```

```python
WARNING: tokenization mismatch: 102 vs. 104. (ignored)
WARNING: tokenization mismatch: 102 vs. 104. (ignored)
WARNING: tokenization mismatch: 129 vs. 131. (ignored)
WARNING: tokenization mismatch: 102 vs. 104. (ignored)
WARNING: tokenization mismatch: 129 vs. 131. (ignored)
WARNING: tokenization mismatch: 129 vs. 131. (ignored)
WARNING: tokenization mismatch: 129 vs. 131. (ignored)
WARNING: tokenization mismatch: 129 vs. 131. (ignored)
WARNING: tokenization mismatch: 129 vs. 131. (ignored)
WARNING: tokenization mismatch: 111 vs. 113. (ignored)
```

loss is ```shell
{'loss': 0.0, 'learning_rate': 4.097492052278347e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 4.2387848816672555e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 4.3800777110561644e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 4.5213705404450727e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 4.6626633698339816e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 4.80395619922289e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 4.945249028611799e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 5.086541858000706e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 5.227834687389614e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 5.369127516778524e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 5.510420346167432e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 5.651713175556341e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 5.793006004945249e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 5.934298834334157e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 6.075591663723067e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 6.216884493111975e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 6.358177322500883e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 6.499470151889792e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 6.6407629812787e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 6.78205581066761e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 6.923348640056518e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 7.064641469445425e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 7.205934298834334e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 7.347227128223243e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 7.488519957612151e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 7.62981278700106e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 7.771105616389968e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 7.912398445778878e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 8.053691275167787e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 8.194984104556694e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 8.336276933945602e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 8.477569763334511e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 8.618862592723419e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 8.760155422112329e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 8.901448251501236e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 9.042741080890145e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 9.184033910279054e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 9.325326739667963e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 9.46661956905687e-07, 'epoch': 0.0}
{'loss': 0.0, 'learning_rate': 9.60791239844578e-07, 'epoch': 0.0}
```

**Comment by Leon1207** [comment]
I have the same problem, could you please share me how to fix it?

**Comment by nixsui** [comment]
same problem

**Comment by lukashelff** [comment]
i am facing the same problem

**Comment by prashanthsadasivan** [comment]
I ran into this yesterday. managed to fix it with advice from this  issue https://github.com/haotian-liu/LLaVA/issues/661#issuecomment-1778630921 . depending on if you're finetuning different models you'll need to adjust the calculations there. for me, I was using the `</s>` separator with tiny llama, so i needed to start `cur_len = 1` and make `cur_len += round_len + 1` to account for the separator tokenization in process_mpt

**Comment by ticketlearn** [comment]
face the same problem

**Comment by Bleking** [comment]
Plus, make sure to set the version value as 'mpt' due to the separation token.
